x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/AbstractNativeProcess.java

```diff
@@ -248,39 +260,43 @@ public ZonedDateTime getProcessStartTime() {
     @Override
     public boolean isProcessAlive() {
         // Sanity check: make sure the process hasn't terminated already
-        return !cppLogHandler.hasLogStreamEnded();
+        return !cppLogHandler().hasLogStreamEnded();
     }
 
     @Override
     public boolean isProcessAliveAfterWaiting() {
-        cppLogHandler.waitForLogStreamClose(Duration.ofMillis(45));
+        cppLogHandler().waitForLogStreamClose(Duration.ofMillis(45));
         return isProcessAlive();
     }
 
     @Override
     public String readError() {
-        return cppLogHandler.getErrors();
+        return cppLogHandler().getErrors();
     }
 
     protected String jobId() {
         return jobId;
     }
 
     protected InputStream processOutStream() {
```

Comment:
Could we also have a similar method for input stream for consistency?
`private OutputStream processInStream();`